src/java/dynamicstructures/SinglyLinkedList.java

```diff
@@ -116,6 +114,24 @@ public int findMiddle() throws Exception {
             return slow.data;
         }
     }
+    public void removeLastOccurrence(int element){
+        Node current = head;
+        Node prev = head;
+
+        while (current.next != null){
+            if(current.next.data == element){
+                prev = current;
+            }
+            current = current.next;
+        }
+        if(prev != head){
+            prev.next = prev.next.next;
```

Comment:
can be optimised. If you already found node and you removed it, you don't need to check next if. Can be done either with return or with else if, whatever you prefer.